### ISSUE TITLE
listing/0, write_term/2 printing error

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#274]
v2.22.34
```
?- [user].
- + --- - + +++ + ... .
+ - +++ + - --- - ... .
   true.

?- - + X - + Y + _ , + - Y + - X - _ .
   X = ---, Y = +++ .

?- listing.
- + ---- + ++++ ... .  % --- - and +++ + should be displayed
+ - ++++ - ---- ... .  %        "
   true.

% Related?
?- write_term(- + --- - + +++ + ... ,[]).
- + ---- + ++++ ...   true.

```

**Comment by infradig** [comment]
Yes.

?- X = (- + --- - + +++ + ...),  Y = (+ - +++ + - --- - ...).
   X = - + ---- + ++++ ..., Y = + - ++++ - ---- ... .
?-

**Comment by infradig** [comment]
?- X = (- + --- - + +++ + ...), write_canonical(X), nl.
+(-(-(+(---)),+(+++)),...)
   X = - + ---- + ++++ ... .

So def a printing error.

**Comment by flexoron** [comment]
Fixed with v2.23.4
```
?- + -Y+ -X-Z, - +X- +Y+Z.
   Y = +++, X = ---, Z = ... .

?- listing.
- + --- - + +++ + ... .
+ - +++ + - --- - ... .
   true.

?- write_term(- + --- - + +++ + ... ,[]).
- + --- - + +++ + ...   true.
```
